spec/05-classes-and-objects.md

```diff
@@ -435,13 +435,13 @@ An early definition is type-checked and evaluated in the scope which
 is in effect just before the template being defined, augmented by any
 type parameters of the enclosing class and by any early definitions
 preceding the one being defined. In particular, any reference to
-`this` in the right-hand side of an early definition refers
+`this` on the right-hand side of an early definition refers
```

Comment:
Early defs are famously convoluted and not long for this world. The incorrect part here is "right-hand side".
```
object X {
  object r { def unapply(s: String) = Some(s.init) }
  object Y extends { val this.r(z) = "hello" } with AnyRef { def f = z }
}
```